odoo/migration/migration.py

```diff
@@ -0,0 +1,2383 @@
+                   FROM information_schema.columns
+                  WHERE table_name=%s
+                    AND column_name NOT IN %s
+               """.format(select=select), params + [table, ignore])
+    return list(zip(*cr.fetchall()))
+
+
+def get_depending_views(cr, table, column):
+    """Get the list of SQL views depending on the specified column."""
+    # http://stackoverflow.com/a/11773226/75349
+    q = """
+        SELECT distinct quote_ident(dependee.relname)
+        FROM pg_depend
+        JOIN pg_rewrite ON pg_depend.objid = pg_rewrite.oid
+        JOIN pg_class as dependee ON pg_rewrite.ev_class = dependee.oid
+        JOIN pg_class as dependent ON pg_depend.refobjid = dependent.oid
+        JOIN pg_attribute ON pg_depend.refobjid = pg_attribute.attrelid
+            AND pg_depend.refobjsubid = pg_attribute.attnum
+        WHERE dependent.relname = %s
+        AND pg_attribute.attnum > 0
+        AND pg_attribute.attname = %s
+        AND dependee.relkind='v'
+    """
+    cr.execute(q, [table, column])
+    return map(itemgetter(0), cr.fetchall())
+
+
+def drop_depending_views(cr, table, column):
+    """Drop views depending on a column.
+
+    This is usually used to ensure that modifying fields will not make
+    SQL views (auto=False models) crash after the upgrade, forcing them
+    to be regenerated during the uprade.
+    """
+    for v in get_depending_views(cr, table, column):
+        cr.execute("DROP VIEW IF EXISTS {0} CASCADE".format(v))
+
+
+def get_fk(cr, table):
+    """Get the list of foreign keys pointing to `table`
+
+        :rtype: list(tuple)
+        :return: [(foreign_table, foreign_column, constraint_name, on_delete_action)]
+                 where on_delete_action if one of the following:
+                 - a: no action
+                 - r: restrict
+                 - c: cascade
+                 - n: set null
+                 - d: set default
+    """
+    q = """SELECT quote_ident(cl1.relname) as table,
+                  quote_ident(att1.attname) as column,
+                  quote_ident(con.conname) as conname,
+                  con.confdeltype
+             FROM pg_constraint as con, pg_class as cl1, pg_class as cl2,
+                  pg_attribute as att1, pg_attribute as att2
+            WHERE con.conrelid = cl1.oid
+              AND con.confrelid = cl2.oid
+              AND array_lower(con.conkey, 1) = 1
+              AND con.conkey[1] = att1.attnum
+              AND att1.attrelid = cl1.oid
+              AND cl2.relname = %s
+              AND att2.attname = 'id'
+              AND array_lower(con.confkey, 1) = 1
+              AND con.confkey[1] = att2.attnum
+              AND att2.attrelid = cl2.oid
+              AND con.contype = 'f'
+    """
+    cr.execute(q, (table,))
+    return cr.fetchall()
+
+
+def delete_unused(cr, table, xmlids, set_noupdate=True):
+    """
+    Delete all records in the provided list not being referenced in any FKEY.
+
+    Note that the xmlids th
```

Comment:
DRY. This already exists: https://github.com/odoo/odoo/blob/d6ee2abb987016c28e24f471f4b3d413af84c185/odoo/sql_db.py#L406-L418